### ISSUE TITLE
Should secondary literature have more than one template?

### ISSUE THREAD

**Issue by fjorba** [rism-digital/muscat#1013]
Currently, in Muscat there are different templates for sources, because different document types require different fields (Marc tags and subfields) and defaults.  I think that secondary literature should have the same capability, because the document types are so different that they require different fields (for example: a journal article and a book monograph).

I volunteer to prepare the infrastructure for that, now that the two models are aligned (#987).  My proposal is to start with a list of a single document type, the current one (we can call it Default, to start with), but that would open the door to easily create others.

What do you think?

**Comment by MFalletta** [comment]
Great idea! I agree

**Comment by jenniferward** [comment]
Me too!
If it helps, I made a very short guide (aimed at Muscat contributors) that talks about the different types of secondary literature in Muscat. It briefly explains fields needed for different types of materials - it might give you some ideas:
http://www.rism.info/fileadmin/content/community-content/Muscat_EN/Secondary_lit_brief_guide.pdf

**Comment by fjorba** [comment]
Thank you, I'll take it into account.

**Comment by xhero** [comment]
I think it is doable, you would add a record_type like in source right?

**Comment by fjorba** [comment]
My idea was to mimick source model as much as possible, with the obvious changes (composer to author, for example). 

> I think it is doable, you would add a record_type like in source right?

I have a prototype working in my local machine, but now I don't remember the details, it was just a proof of concept that I could do it.  It was running some weeks ago, I'll have to refresh my memory.  But yes, if you count on record_type being there, I'll double check that it is there.

Obviously I'll require your review, and I'll appreciate your comments.

**Comment by ahankinson** [comment]
A heads-up that there are some changes coming for languages (#1004) that will probably affect this work, since (I believe) you will need a "Labels" file that gives the translations for each field.

**Comment by fjorba** [comment]
What a nice moment for such a change!  (Yes, I'm sure it is never a good moment for such a change...)

Do you consider that I should your patch before I publish mine, so I can understand how it affects the rest of the cataloguing files?  My idea is to prepare the infrastructure for choosing templates for 7.0.  If I can create a new one (maybe starting with journal article), great, but that would be a second step.

**Comment by xhero** [comment]
I think at the end it should not really change that much on the implementation side of things for the templates, as the languages are managed (more or less) transparently in the backend.
To make it work as source, you will probably need to add a record_type attribute to keep information on the proper template, and then as in source select the right tags to display (all of them should exist in the tag configuration). You will also need to make the appropriate "empty" templates used when creating a new source. I think that should be it!

**Comment by fjorba** [comment]
Thanks, I haven't really started yet, but I'll keep that in mind.

However, I do have a first question: would you prefer a single, unified, record_type list for both bibliographic (sources or secondary literature) records, or two lists?  Intuitively, I would say a single one, like there is a single authority record list for persons, institutions, etc.  However, I'd probably say that it should be a configuration file, not a model, db table and so on, as it is closely tied to configuration files, and so it belongs to installation data, not to (user supplied) bibliographic information.

**Comment by xhero** [comment]
I'm not sure I follow completely. ```record_type``` is used to attach a template to a source, and a template being basically a selection and grouping of marc tags. Since marc specification for authority files and bibliographic records is different, it seems to me the templates are something that should be tailored to each particular model they are used on.

**Comment by fjorba** [comment]
I was referring to (re)use the same RECORD_TYPES list in https://github.com/rism-ch/muscat/blob/develop/lib/marc_source.rb.  Should I?

I digged somewhat into this list when debugging #934, but then I found some redundancy with default_mapping in https://github.com/rism-ch/muscat/blob/develop/config/marc/default/source/template_configuration.yml#L60.

**Comment by jenniferward** [comment]
Something related but separate: Could you @fjorba perhaps take a look at Secondary Literature's field 520 in the context of your activities here? It's called "Contents note" but this is incorrect. The field is meant to describe dates of publication, like for serials, and indeed that's what it has been used for if you look at the data (example: https://muscat.rism.info/admin/catalogues/50005776). This corresponds to MARC 362 instead.

**Comment by fjorba** [comment]
Is this change from 520 tags to 362 (I agree about it, but 520 should also exist) like the one from 240 to 245 in #626 (I also agree), @jenniferward?

I don't know how do you handle those corrections, and how do you coordinate Muscat data with the public RISM opac, and I feel very insecure about proposing anything. However, a possible path to correct those tags could be:

1. Create the infrastructure to allow different templates (a future pull request to be linked to this issue).
2. Have a single default template (the current one), to start with.  All current records stay here, with the corresponding record_type for this default template (probably in a the same pull request above).
3. Create a second, specific template. I would start with journal article, because it has less variability.
4. Change journal articles from the default record_type to this future journal article record type.  At the same time, change the dubious tags to better tags (current 240 to 245, and 520 to 362, for example).  That should have a specific issue, of course (related to #626).  I don't think that I can do it, probably it should be done with a maintenance script run by your team.
5. Create a third template, maybe monograph.
6. Do the same changes with monograph records. 
7. Continue until all secondary literature records have their specific record type and their tags corrected.

But, again, this may not be the way you work at the ZR and the Muscat develoment team, so I'm just guessing.  So sorry if I'm completely wrong with my proposal.

**Comment by jenniferward** [comment]
To address the first part of your question: Yes, it's similar in nature in that the tag is incorrect.
Not to further complicate matters, but I should point out that 520 is called "Contents note" in the Secondary Literature, but "Contents note" in MARC is more correctly 505. A 520 note would be more of a summary. Both 505 and 520 would likely be useful for publications, though.

**Comment by fjorba** [comment]
Thank you, @jenniferward.  Maybe this new template mechanism could be a good moment to review the current used tags and subfields for secondary literature?  For example, besides the tags you have mentioned, for journal articles, the title of the journal, issn, related parts should go to 773, as stated in the standard (https://www.loc.gov/marc/bibliographic/bd773.html) rather than the current 760 (https://www.loc.gov/marc/bibliographic/bd760.html).  The LoC examples are clear enough.  760 could be used for monographs (although we don't use it in any of our systems).

I know that changes tags affect many other parts, like indexing, display, etc., but what I'd like to know if you all think that the right moment can be now, as we create specific templates for different document types.

I don't know either (I haven't had the time to investigate it) if this use of 773 in secondary conflicts with the current use in sources (that I suspect that should be 760, but that's another story).

**Comment by xhero** [comment]
> I was referring to (re)use the same RECORD_TYPES list in https://github.com/rism-ch/muscat/blob/develop/lib/marc_source.rb. Should I?
> 
> I digged somewhat into this list when debugging #934, but then I found some redundancy with default_mapping in https://github.com/rism-ch/muscat/blob/develop/config/marc/default/source/template_configuration.yml#L60.

I don't think we should necessarily reuse those templates, as they are very specific to musical sources. At the end a template is really just a list of tags to show/not show for a particular record and they are tailored to the different data types.
The ```default_mapping``` maps the template to the template file. It used to me automagic but we decided it was less confusing to have the mapping explicit. The names ```000_collection``` are the template files to read.

**Comment by jenniferward** [comment]
Hi @fjorba: I think of this ticket from time to time. We need to make it easier for people to enter Secondary Lit, because I think the current scheme (one template for everything) prevents some people from entering anything at all, and we end up with unstructured Lit references in Sources and People. 

The template makes it especially difficult to enter web resources (such as a database, or even a website), and validations or required/recommended fields are impossible due to the varied nature of what we enter here.

Not sure where to begin, or if this is still relevant to you. What are your current thoughts?

**Comment by fjorba** [comment]
Yes, this issue is relevant for us, and we have a local patch to implement it in our local instances. There is an obsolete version at #1043, but we keep updating it for each new Muscat release.

In fact, all relevant infrastructure is already in Muscat for sources, and basically the work consists in adapt it to publications. What made it difficult (for me, at least) was to understand the complex rules you have for sources. In the case of publications, we decided a simple one-to-one rule between document type and template. For Traces, there is the classic distinction between books, journals, articles and chapters; for DDD we have many more (videos, theses, student guides, audios, etc), but we still keep a one-to-one correlation.

Again, like in Muscat sources, each publication "template" is a subset of the list of the whole tags allowed. I'll post our current version in the next few days.

**Comment by xhero** [comment]
I don't know, I'm still not a great fan of this. While technically it is possible (and not even too complex to implement), I'm unsure this is the right thing to do. With Sources, that it is our main bibliographic focus, but publications are really just references to other stuff. This also opens all the problems we have with templates in Sources to Publications: people misusing templates or randomly selecting them, then we need a mechanism to migrate templates when they will be invariably wrong in Publications, and so on.
What are the problems that people are facing? Wouldn't it be enough to restructure the Publications page. For example digital resources, what are the problems that users have here? Entering 856 should be quite straightforward no?

**Comment by fjorba** [comment]
In our case, the "migration" is just changing the record type, because there is a one-to-one relation. If, for example, you leave the record type editable (even if limited to editor or admin), changing a record type of a record (say, a chapter to an article) would be trivial. (For historical reasons and compatibility with our current Invenio system, we have a different way of coding this record type for each of those systems; but it is just a value in a bibliographic tag.)

But yes, having different templates is important, and sure it would make the catalogers life easier, even if they are "just" references in Muscat. An ISSN goes to 022 $a for a journal, but to 773 $x for an article. In our older records we had too many errors for this kind of confusions, and it is frustrating. The same with ISBN (020 $a vs 773 $z) or DOIs, and those identifications are specially common and important. Confusions about authors and titles do also exist in component parts if it is not clear that you are cataloguing a component part (say, a chapter).

**Comment by jenniferward** [comment]
@fjorba We discussed this internally and we are going to try a somewhat modified procedure where the selection of a "type" from the Secondary Literature template triggers a set of recommended (or required) fields. On a small scale I have already outlined this in my guidance for our catalogers using Secondary Literature: 

https://rism.info/resources-old-website/community-content/Muscat_EN/Secondary_lit_brief_guide.pdf

So a cataloger would choose a publication type (maybe based on our 337 or the 240$g/h) and then be prompted to fill out only the relevant fields. (I'll write a separate ticket for that.)

In a way, it might be considered a modified template, but we are going to try to support it through validations rather than separate templates. You are right that things can land in the wrong fields and some fields are completely irrelevant for certain kinds of publications, so I hope that a publication type can help the cataloger focus on the correct fields.

A side note: As you know, the Sources module is our focus. Supporting our source descriptions and authorities with evidence from secondary literature is important, but it's a funny thing because secondary literature is very frequently described in other bibliographic databases and we find ourselves recreating a lot in Muscat. Generally we are hampered by the structure of this Secondary Literature module. For example, for a cataloger to cite a book chapter, they have to enter the chapter, the book, and potentially book series (3 records) plus potentially 2 or more new name authorities (for the author and editor(s) of publications), all of which takes up valuable time resources when all you wanted to state in the first place was evidence for a composer's first name. Looking ahead, it would be more efficient to be able to integrate or link out to other bibliographic systems without needing to recreate it all in Muscat. We don't have a solution for that right now, but the Secondary Lit is a strange animal for us (or me, this is my opinion).

**Comment by fjorba** [comment]
I understand, thanks for the explanation. But related to that, I have another request, that I would love to have: standard titles for publications. It turns out that for publications there are relations to people, institutions, places, standard terms, but not standard titles.

Even for your more restricted usage, I think that you would benefit for it (journal titles, collective books, etc.). That would allow autocomplete for those titles/journals, etc. May we have this, @xhero? I can work on a PR.

**Comment by jenniferward** [comment]
I think we use standard titles a bit differently than you (and recall our MARC error with 240: https://github.com/rism-digital/muscat/issues/626). Part of our complicated system is that you create a separate record for the journal or the collective book. 

In the field 760, you enter the title of the book or journal, then it makes the link:

<img width="661" height="141" alt="Image" src="https://github.com/user-attachments/assets/e3328dc8-fa30-4db2-a6d0-3966be00079b" />

<img width="1174" height="259" alt="Image" src="https://github.com/user-attachments/assets/ba5ebf18-422b-4dbf-9b8d-0e01de0550ea" />

So in that sense, we don't need a standardized title in our setup, because you are linking to the journal or book record. Or maybe there is an aspect I'm not thinking of?

If we move 240 to 245, then that would free up 240 for you if you want for any standardizing (we would not use it, I dare say; but see https://github.com/rism-digital/muscat/pull/1363).

**Comment by fjorba** [comment]
I see. As far as understand from your screenshot and from https://github.com/rism-digital/muscat/blob/develop/config/marc/tag_config_publication.yml#L423, you are linking publications to publications, not to standard titles.

I have checked and we do not have this relation. But we have lots of relations from publications to standard titles. As the UAB official repository, we hold the (open version) of the UAB research output, and that means hundreds, thousands of journals that we don't have the record (publication) of, and we do not pretend to have it. They are just standard titles, and we reuse other standard Muscat fields for ISSN, and which kind of title is it and some other information. Standard titles are also meeting names, book series, book titles (when cataloging a chapter), etc, but specially many university related materials: course names, degrees (all them using the standard UAB ids), etc. We do have the full publication record of the journals published by our university, but the relation with its articles is via the journal standard title, not publication-to-publication like you.

>  For example, for a cataloger to cite a book chapter, they have to enter the chapter, the book, and potentially book series (3 records) plus potentially 2 or more new name authorities (for the author and editor(s) of publications), all of which takes up valuable time resources when all you wanted to state in the first place was evidence for a composer's first name. 

Maybe using this standard title usage that we practice at UAB would make your citations work easier. So you don't have to catalog the whole Journal of This and That, because it is just a title (with its ISSN, we put is in the alternate name field), and then you refer to it. Or the Encyclopedia of Muscal Whatever, it is just another title (with its ISBN, also as alternate name in our case). You refer to the component (article) part, and that is much lighter than have both records.

So far, we are using an ugly internal trick so that those publications-to-standard-titles relations work, but as they are not backed by Rails internal links, if any of those titles disappear, the application crashes.  The real solution would be to make a proper Rails relation, and this is one of the pending issues. That's why, if it does not harm RISM, I'd like to make this relation. What's your position, @xhero? Should I open another issue for it so we can discuss it properly?

**Comment by xhero** [comment]
I think the Standard Titles in RISM should be of actual RISM sources, if we add publications then we will have stuff like

Masses
Sacred songs
Auf dass erfüllet würde die Schrift
Das "Große Hymnus-Buch" des Kantors Christoph Schultze
Der Marktflecken Bibra : eine Darstellung seiner politischen und kirchlichen Entwickelung
Lieder, Frauenst., op. 76 [Zwiegesang]

In the same bucket. It is already quite unclear because we put there Standard Title and Incipit texts...

**Comment by jenniferward** [comment]
Yeah that might simplify things in the long run, but our Standard Titles are a bit of a mess right now and we can't add another layer. 
Though sometimes we add URLs and other info, so the full(er) records have some advantages. 
But I don't have anything against @fjorba adding it for his own use.

**Comment by fjorba** [comment]
> I think the Standard Titles in RISM should be of actual RISM sources, if we add publications then we will have stuff like

I see. However, the confusion only would happen if there is a proper `foreign_class` entry in https://github.com/rism-digital/muscat/blob/develop/config/marc/tag_config_publication.yml.

What I don't know (or I'm still unsure) is what happens if I have a different db schema than upstream Muscat, but I'd prefer not having to, just in case (Rails is Rails!). Having another table with this relation, empty in your case, would harm you? Of course you'd prefer not, but that would be a big win for us.
